### DISCUSSION TITLE
A way to totally block Contact form?

### DISCUSSION THREAD

**Discussion by eurobank** [mindstellar/shopclass#431] [General]
The last week we have tones of malicious registered users to contact other users with spam, via their ads contact form. Is there a way to totally disable Contact from ads for ALL users (i can see only for visitors and not for registered users)?

**Comment by navjottomer** [comment]
Right now I am in medical emergency will look back to it after 2-3 days.

**Comment by eurobank** [comment]
Sorry to hear that, i wish all the best.

**Comment by eurobank** [comment]
Today again someone made an account and send around 200 scam/loans contact form emails via the Item contact page. 

Some user that got the email, complained to our hosting company and since it was the second time that happened, i LOST my smtp access for that site. 

And now i have to prove that i somehow fixed this issue (so i get back my smtp access) but i can't do that. How to do i fix it? There is NO way that i know of.

**Comment by navjottomer** [comment]
You are talking about main contact form on website?

**Reply by eurobank** [reply]
No, from a LOGGED/REGISTERED user ITEM contact form (sidebar). Scammers make an account and use that to contact the posters.

**Comment by eurobank** [comment]
@navjottomer 

We have talked in private with some webmasters running Osclass and we have come up with the only solution for this issue. 

The advertizer should have his Contact emails ONLINE IN osclass dashboard and not getting an email with the Contact body. Kind of a mailbox or something much much simpler. He could only get an email telling him that he got a new Contact form item.

But this doesn't exist in Osclass right now. 

If you have any other ideas please share.

**Comment by eurobank** [comment]
An other idea is to enable some kind of slow down, as we have for the ads posting.

**Comment by navjottomer** [comment]
Having a private mailbox is a good idea but will require lots of effort and we cannot go in that direction right now.
For the throttling of email posts, this can be done easily, we can store user activity in the database and can provide an option to restrict users to certain numbers of emails/day or hour.
For right now if you want to disable the item-contact form. Then you can put this code in your theme functions.php file
```php
$disableItemContactForm = function ($item){
    osc_add_flash_error_message( __( 'Contact form is disabled temporarily, please try posting a comment on listing' , 'osclass' ) );
    osc_redirect_to( osc_item_url() );
}
osc_add_hook('pre_item_contact_post', $disableItemContactForm);
unset($disableItemContactForm);
```
Please test the above code as I have not tested it.

**Comment by eurobank** [comment]
@navjottomer I will test and let you know. 

Also can the contact emails be somehow moderated? before sent out?

**Comment by eurobank** [comment]
Maybe in the future someone can make a plugin, ONLY for the Contacts to be moderated. Not a full email box. Just for Contact form emails, like it was in the legacy (and not working now) Antispam plugin from Liath.

**Comment by eurobank** [comment]
Thanks @navjottomer the above drop in fuction seems to work.

A related question: i have removed the code Send a friend from all pages. BUT the page "item/send-friend/31124" for example are available if you directly point to it. So easy to find.

How can i also disable the Send a Friend?
